graphics/fonts/ttf.cpp

```diff
@@ -594,16 +594,25 @@ static void renderGlyph(uint8 *dstPos, const int dstPitch, const uint8 *srcPos,
 			if (*src == 255) {
 				*rDst = color;
 			} else if (*src) {
-				const uint8 a = *src;
+				sA = *src;
 
-				uint8 dR, dG, dB;
-				dstFormat.colorToRGB(*rDst, dR, dG, dB);
+				uint8 dA, dR, dG, dB;
+				dstFormat.colorToARGB(*rDst, dA, dR, dG, dB);
 
-				dR = ((255 - a) * dR + a * sR) / 255;
-				dG = ((255 - a) * dG + a * sG) / 255;
-				dB = ((255 - a) * dB + a * sB) / 255;
+				if (dA == 0) {
+					*rDst = dstFormat.ARGBToColor(sA, sR, sG, sB);
+				} else {
+					double alpha = (double)sA / 255.0;
+					dR = static_cast<uint8>((sR * alpha) + (dR * (1.0 - alpha)));
+					dG = static_cast<uint8>((sG * alpha) + (dG * (1.0 - alpha)));
+					dB = static_cast<uint8>((sB * alpha) + (dB * (1.0 - alpha)));
 
-				*rDst = dstFormat.RGBToColor(dR, dG, dB);
+					if (sA > dA) {
+						dA = static_cast<uint8>((sA * alpha) + (dA * (1.0 - alpha)));
+					}
+
```

Comment:
Where do those formula come from?
My understanding of color blending is that it should be:
```c++
double sAn = (double)sA / 255.0;
double dAn = (double)dA / 255.0;
dR = static_cast<uint8>(sR * sAn + dR * dAn * (1.0 - sAn) / (sAn + dAn * (1.0 - sAn)));
dG = static_cast<uint8>(sG * sAn + dG * dAn * (1.0 - sAn) / (sAn + dAn * (1.0 - sAn)));
dB = static_cast<uint8>(sB * sAn + dB * dAn * (1.0 - sAn) / (sAn + dAn * (1.0 - sAn)));
dA = static_cast<uint8>(sA + dA * (1.0 - sAn));
```
This also means there is no need for a special case if `dA` is 0.
And you can of course optimise that code a bit by computing `dAn * (1.0 - sAn)` only once instead of 6 times.

Reply:
Mostly followed suit with managed_surface.cpp:487-490
Figured if case for dA is 0 might be better to avoid calculations, but perhaps not.
Destination alpha... I just wasn't sure the best call.

Reply:
OK, looking at https://en.wikipedia.org/wiki/Alpha_compositing#Alpha_blending
If I'm understanding it correctly, what I used would be correct for dA being 255, but not complete enough for any other value.
I'll take another swing at this.

Reply:
Yep, the suggested code was great. 
I went with calculating a normalized output alpha and converting that a byte at the end.
I thought it would be a little more clear to anyone reading the code.